### PR TITLE
feat: fix sonar findings

### PR DESCRIPTION
**Pull request by rjaegers**

### CODE REVIEW

File: protobuf/echo/Echo.cpp [hunk 1]
```diff
@@ -45,9 +45,8 @@ namespace services
         return echo;
     }
 
-    ServiceProxy::ServiceProxy(Echo& echo, uint32_t id, uint32_t maxMessageSize)
+    ServiceProxy::ServiceProxy(Echo& echo, EMIL_MAYBE_UNUSED uint32_t serviceId, uint32_t maxMessageSize)
         : echo(echo)
-        , serviceId(id)
```
Comment (richardapeters) [inline protobuf/echo/Echo.cpp:50]: this looks like a mistake. id should be used to initialize serviceId

File: protobuf/echo/Echo.hpp [hunk 2]
```diff
@@ -198,7 +199,7 @@ namespace services
         : public infra::IntrusiveList<ServiceProxy>::NodeType
     {
     public:
-        ServiceProxy(Echo& echo, uint32_t id, uint32_t maxMessageSize);
+        ServiceProxy(Echo& echo, EMIL_MAYBE_UNUSED uint32_t serviceId, uint32_t maxMessageSize);
```
Comment (richardapeters) [inline protobuf/echo/Echo.hpp:202]: This is just the declaration; does it need an [[unused]] tag as well?

File: protobuf/echo/Echo.hpp [hunk 3]
```diff
@@ -207,7 +208,6 @@ namespace services
 
     private:
         Echo& echo;
-        uint32_t serviceId;
```
Comment (richardapeters) [inline protobuf/echo/Echo.hpp:210]: if serviceId is not necessary, then the parameter to the constructor should be removed completely

File: services/network/BonjourServer.hpp [hunk 4]
```diff
@@ -51,7 +51,6 @@ namespace services
 
         private:
             BonjourServer& server;
-            uint16_t queryId;
```
Comment (richardapeters) [inline services/network/BonjourServer.hpp:54]: if queryId is not used, then it should be removed from the constructor

File: services/tracer/TracerWithTime.cpp [hunk 5]
```diff
@@ -10,12 +11,10 @@ namespace services
 
     void TracerWithTime::InsertHeader()
     {
-        infra::TimePoint nowTimePoint = infra::Now();
-        time_t now = std::chrono::system_clock::to_time_t(nowTimePoint);
-        std::tm* utcTime = gmtime(&now);
```
Comment (richardapeters) [inline services/tracer/TracerWithTime.cpp:15]: Ouch I thought I had gotten rid of gmtime and friends a long time ago. Thanks!

**Review by richardapeters — APPROVED** [review]

File: protobuf/echo/Echo.hpp (hunk 2 above)
Comment (rjaegers) [reply protobuf/echo/Echo.hpp:202]: Probably not, no. If we keep the parameter, I will remove this.

File: protobuf/echo/Echo.hpp (hunk 3 above)
Comment (rjaegers) [reply protobuf/echo/Echo.hpp:210]: I have removed it completely. Including from the code generator.

File: services/network/BonjourServer.hpp (hunk 4 above)
Comment (rjaegers) [reply services/network/BonjourServer.hpp:54]: queryId is not used as a member in the class, but it is used in the contstructor to fill in the queryId of DnsRecordHeader, a temporary that is then streamed to stream.

File: services/tracer/TracerWithTime.cpp (hunk 5 above)
Comment (rjaegers) [reply services/tracer/TracerWithTime.cpp:15]: There is still one instance left... I'll see if I can remove that as well.

**Pull request merged**